addon/modifiers/container-query.ts

```diff
@@ -131,11 +133,13 @@ export default class ContainerQueryModifier<
   private evaluateQueries(): void {
     const queryResults = {} as QueryResults<T>;
 
-    for (const [featureName, metadata] of Object.entries(this.features)) {
-      const { dimension, min, max } = metadata as Metadata;
+    for (const [featureName, metadata] of Object.entries(
+      this.features
+    ) as ObjectEntries<Features<T>>) {
```

Comment:
We defined `this.features` (i.e. an external source did not). As a result, we should be able to assert that `Object.entries(this.features)` has the type of `ObjectEntries` instead of [`type-fest`'s broader type, `Entries`](https://github.com/sindresorhus/type-fest/blob/main/source/entries.d.ts#L57-L62).